lambdas/es/indexer/index.py

```diff
@@ -487,13 +487,19 @@ def retry_s3(
     else:
         arguments['IfMatch'] = etag
 
-    time_remaining = get_time_remaining(context)
+    def not_known_exception(exception):
+        error_code = exception.response.get(['Error'], {}).get(['Code'], 218)
+        return error_code not in ["402", "403", "404"]
```

Comment:
What error codes do we expect to see and retry? In general, no 4xx codes should be re-tried...

Reply:
That seems a bit extreme. e.g. we want to retry 429s (too many requests).

Reply:
Seems more correct to enumerate the 400s we do want to retry (it's true most aren't worth trying) there are only a few to retry; see Client Errors https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Reply:
I think 408, 409, and 429 are the only ones worth retrying.